src/Microsoft.Tye.Hosting/DockerImagePuller.cs

```diff
@@ -66,16 +64,34 @@ private async Task PullContainerAsync(string image)
         {
             await Task.Yield();
 
+            bool installed = false;
+            var result = await ProcessUtil.RunAsync(
+                                    "docker",
+                                    $"images --filter \"reference={image}\" --format \"{{{{.ID}}}}\"",
+                                    outputDataReceived: data => installed = true,
+                                    throwOnError: false);
+
+            if (result.ExitCode != 0)
+            {
+                throw new CommandException("Docker images command failed");
+            }
+
```

Comment:
When the image is not installed it will continue to pull the images. This command

    images --filter "reference={image} --format "{{.ID}}"

only output the single line that match with the filter condition. In such case I set the installed flag to true.

     outputDataReceived: data => installed = true,

for instance:

    >docker images --filter "reference=jboss/keycloak:6.0.1" --format "{{.ID}}"
    3a6718ca4ee0

I don't print the output, I just catch it to set the flag to true.

Reply:
Though this is probably fine, it'd be more clear if we did a few things:
- Printed the output of this command images in case of failure.
- Get the output in the result and check if it's non-empty to determine if it's installed (remove the outputDataReceived callback and use result.Output)

Reply:
Fixed